openwisp_radius/tests/test_api.py

```diff
@@ -2570,3 +2568,25 @@ def test_user_phone_number_unique(self):
             }
         )
         self._test_phone_number_unique_helper('+23767779235')
+
+    def test_handle_phone_token_validation(self):
+        User.objects.create_user(
+            username='user1',
+            email='email1@gmail.com',
+            password='pass',
+            phone_number='+23767779235',
+        )
+        self._create_user_helper(
+            {
+                'username': 'user2',
+                'email': 'user2@gmail.com',
+                'password1': 'password',
+                'password2': 'password',
+                'phone_number': '+237674479231',
+            }
+        )
+        user = User.objects.get(email='user2@gmail.com')
+        user.is_active = True
+        user.save()
+        self._test_phone_number_unique_helper('+23767779235')
+        self.assertTrue(user.is_active)
```

Comment:
I am not sure if this test covers a case that it was not already tested before, are you sure it's needed?
We already got to almost 2600 lines of code in this test file, we'll have to split it, but for the moment we can start double checking if new tests are really needed.

Reply:
Yes @nemesisdesign . The test is to ensure that when there is a problem during phone number change, the state of the user is not changed. There's no test  for this reason why I added it

Reply:
Nor the test name nor the test code reflect that. Please make it more explicit and easy to read. Where is the error being triggered?